docs/security.md

```diff
@@ -57,6 +57,11 @@ To enable authorization, Spark Master should have
 client should provide HTTP `Authorization` header which contains JSON Web Token signed by
 the shared secret key.
 
+In conjunction to, or alternatively to the JWSFilter based authorisation, you can utilise a proxy to handle authentication to the REST submission server.
+You must configure the setting `spark.master.rest.auth.mode` to `"SecureGateway"`, which will enable you to continue to use the
+setting `spark.authenticate` properties for all other internal authentication, unless specified otherwise below. The default
+setting of this value is `"None"`, which will throw fatal errors during master startup when used in combination with `spark.authenticate`.
```

Comment:
For proxy based authentication, `SecureGateway` is the only auth mode available. It can be enable by setting `spark.master.rest.auth.mode` to `SecureGateway`. Default value for `spark.master.rest.auth.mode` is `None`.